utils/log.go

```diff
@@ -46,42 +42,36 @@ func SetLogLevel(level LogLevel) {
 // SetLogTimeFormat sets the format of the timestamp
 // an empty string disables the logging of timestamps
 func SetLogTimeFormat(format string) {
+	log.SetFlags(0) // disable timestamp logging done by the log package
 	timeFormat = format
 }
 
 // Debugf logs something
 func Debugf(format string, args ...interface{}) {
 	if logLevel == LogLevelDebug {
-		mutex.Lock()
-		logTimestamp()
-		fmt.Fprintf(out, format+"\n", args...)
-		mutex.Unlock()
+		logMessage(format, args...)
 	}
 }
 
 // Infof logs something
 func Infof(format string, args ...interface{}) {
 	if logLevel <= LogLevelInfo {
-		mutex.Lock()
-		logTimestamp()
-		fmt.Fprintf(out, format+"\n", args...)
-		mutex.Unlock()
+		logMessage(format, args...)
 	}
 }
 
 // Errorf logs something
 func Errorf(format string, args ...interface{}) {
 	if logLevel <= LogLevelError {
-		mutex.Lock()
-		logTimestamp()
-		fmt.Fprintf(out, format+"\n", args...)
-		mutex.Unlock()
+		logMessage(format, args...)
 	}
 }
 
-func logTimestamp() {
+func logMessage(format string, args ...interface{}) {
 	if len(timeFormat) > 0 {
-		fmt.Fprintf(out, time.Now().Format(timeFormat)+" ")
+		log.Printf(time.Now().Format(timeFormat)+" "+format, args...)
```

Comment:
This is a red flag from a security point of view (never have dynamic format strings). It doesn't really make a difference right now, but please do this in two prints.